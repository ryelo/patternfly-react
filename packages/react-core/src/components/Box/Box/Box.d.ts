import { HTMLProps } from 'react';

export interface BoxProps extends HTMLProps<HTMLDivElement> {
  children?: React.ReactNode;
}

declare const Box: React.SFC<BoxProps>;

export default Box;
