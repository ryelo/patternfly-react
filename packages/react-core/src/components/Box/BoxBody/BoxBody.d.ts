import { HTMLProps } from 'react';

export interface BoxProps extends HTMLProps<HTMLDivElement> {
  children?: React.ReactNode;
}

declare const BoxBody: React.SFC<BoxProps>;

export default BoxBody;
