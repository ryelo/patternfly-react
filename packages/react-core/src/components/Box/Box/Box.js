import React from 'react';
import { css } from '@patternfly/react-styles';
import PropTypes from 'prop-types';
import styles from '@patternfly/patternfly-next/components/Box/styles.css';

const propTypes = {
  /** content rendered inside the button */
  children: PropTypes.node,
  /** additional classes added to the button */
  className: PropTypes.string
};

const defaultProps = {
  children: '',
  className: ''
};

const Box = ({ children, className, ...props }) => (
  <div {...props} className={css(styles.box, className)}>
    {children}
  </div>
);

Box.propTypes = propTypes;
Box.defaultProps = defaultProps;

export default Box;
