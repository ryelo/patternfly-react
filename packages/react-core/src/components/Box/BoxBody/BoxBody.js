import React from 'react';
import { css } from '@patternfly/react-styles';
import PropTypes from 'prop-types';
import styles from '@patternfly/patternfly-next/components/Box/styles.css';

const propTypes = {
  /** content rendered inside the boxBody */
  children: PropTypes.node,
  /** additional classes added to the boxBody */
  className: PropTypes.string
};

const defaultProps = {
  children: '',
  className: ''
};

const BoxBody = ({ children, className, ...props }) => (
  <div {...props} className={css(styles.boxBody, className)}>
    {children}
  </div>
);

BoxBody.propTypes = propTypes;
BoxBody.defaultProps = defaultProps;

export default BoxBody;
