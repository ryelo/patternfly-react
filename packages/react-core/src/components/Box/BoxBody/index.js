export { default as BoxBody } from './BoxBody';
