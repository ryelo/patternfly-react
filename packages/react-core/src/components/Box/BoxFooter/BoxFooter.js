import React from 'react';
import { css } from '@patternfly/react-styles';
import PropTypes from 'prop-types';
import styles from '@patternfly/patternfly-next/components/Box/styles.css';

const propTypes = {
  /** content rendered inside the boxFooter */
  children: PropTypes.node,
  /** additional classes added to the boxFooter */
  className: PropTypes.string
};

const defaultProps = {
  children: '',
  className: ''
};

const BoxFooter = ({ children, className, ...props }) => (
  <div {...props} className={css(styles.boxFooter, className)}>
    {children}
  </div>
);

BoxFooter.propTypes = propTypes;
BoxFooter.defaultProps = defaultProps;

export default BoxFooter;
