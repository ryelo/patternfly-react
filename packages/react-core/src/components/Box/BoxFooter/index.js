export { default as BoxFooter } from './BoxFooter';
