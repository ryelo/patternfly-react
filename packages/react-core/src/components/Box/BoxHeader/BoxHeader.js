import React from 'react';
import { css } from '@patternfly/react-styles';
import PropTypes from 'prop-types';
import styles from '@patternfly/patternfly-next/components/Box/styles.css';

const propTypes = {
  /** content rendered inside the boxHeader */
  children: PropTypes.node,
  /** additional classes added to the boxHeader */
  className: PropTypes.string
};

const defaultProps = {
  children: '',
  className: ''
};

const BoxHeader = ({ children, className, ...props }) => (
  <div {...props} className={css(styles.boxHeader, className)}>
    {children}
  </div>
);

BoxHeader.propTypes = propTypes;
BoxHeader.defaultProps = defaultProps;

export default BoxHeader;
