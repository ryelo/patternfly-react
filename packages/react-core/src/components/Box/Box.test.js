import Box from './Box/Box';
import BoxHeader from './BoxHeader/BoxHeader';
import BoxBody from './BoxBody/BoxBody';
import BoxFooter from './BoxFooter/BoxFooter';
import React from 'react';
import { shallow } from 'enzyme';

test('Regular Box', () => {
  const view = shallow(<Box />);
  expect(view).toMatchSnapshot();
});

test('Box with Header, Body, Footer', () => {
  const view = shallow(
    <Box>
      <BoxHeader> Box Header </BoxHeader>
      <BoxBody> Box Body </BoxBody>
      <BoxFooter> Box Footer </BoxFooter>
    </Box>
  );
  expect(view).toMatchSnapshot();
});

test('Box with no header', () => {
  const view = shallow(
    <Box>
      <BoxBody> This box has no header </BoxBody>
      <BoxFooter> Box Footer </BoxFooter>
    </Box>
  );
  expect(view).toMatchSnapshot();
});

test('Box with no footer', () => {
  const view = shallow(
    <Box>
      <BoxHeader> Box Header </BoxHeader>
      <BoxBody> Box Body </BoxBody>
    </Box>
  );
  expect(view).toMatchSnapshot();
});

test('Box with only content', () => {
  const view = shallow(
    <Box>
      <BoxBody> Box Body </BoxBody>
    </Box>
  );
  expect(view).toMatchSnapshot();
});
