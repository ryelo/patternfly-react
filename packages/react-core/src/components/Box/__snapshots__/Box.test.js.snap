// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box with Header, Body, Footer 1`] = `
.pf-c-box {
  display: flex;
  flex-direction: column;
  background-color: #ffffff;
  box-shadow: 0 0.0625rem 0.125rem 0 rgba(3, 3, 3, 0.2);
}

<div
  className="pf-c-box"
>
  <BoxHeader
    className=""
  >
     Box Header 
  </BoxHeader>
  <BoxBody
    className=""
  >
     Box Body 
  </BoxBody>
  <BoxFooter
    className=""
  >
     Box Footer 
  </BoxFooter>
</div>
`;

exports[`Box with no footer 1`] = `
.pf-c-box {
  display: flex;
  flex-direction: column;
  background-color: #ffffff;
  box-shadow: 0 0.0625rem 0.125rem 0 rgba(3, 3, 3, 0.2);
}

<div
  className="pf-c-box"
>
  <BoxHeader
    className=""
  >
     Box Header 
  </BoxHeader>
  <BoxBody
    className=""
  >
     Box Body 
  </BoxBody>
</div>
`;

exports[`Box with no header 1`] = `
.pf-c-box {
  display: flex;
  flex-direction: column;
  background-color: #ffffff;
  box-shadow: 0 0.0625rem 0.125rem 0 rgba(3, 3, 3, 0.2);
}

<div
  className="pf-c-box"
>
  <BoxBody
    className=""
  >
     This box has no header 
  </BoxBody>
  <BoxFooter
    className=""
  >
     Box Footer 
  </BoxFooter>
</div>
`;

exports[`Box with only content 1`] = `
.pf-c-box {
  display: flex;
  flex-direction: column;
  background-color: #ffffff;
  box-shadow: 0 0.0625rem 0.125rem 0 rgba(3, 3, 3, 0.2);
}

<div
  className="pf-c-box"
>
  <BoxBody
    className=""
  >
     Box Body 
  </BoxBody>
</div>
`;

exports[`Regular Box 1`] = `
.pf-c-box {
  display: flex;
  flex-direction: column;
  background-color: #ffffff;
  box-shadow: 0 0.0625rem 0.125rem 0 rgba(3, 3, 3, 0.2);
}

<div
  className="pf-c-box"
/>
`;
